Sources/Markdown/Interpretive Nodes/Aside.swift

```diff
@@ -170,51 +180,72 @@ public struct Aside {
+    }
+
+    /// Create an aside from a block quote if it contains a kind tag that matches the given requirement.
+    public init?(_ blockQuote: BlockQuote, tagRequirement: TagRequirement = .tagNotRequired) {
+        guard tagRequirement != .tagNotRequired else {
+            self = .init(blockQuote)
+            return
+        }
+
+        guard let kindTag = blockQuote.parseAsideTag(tagRequirement: tagRequirement) else {
+            return nil
         }
-        initialText.string = String(trimmedText)
 
-        let newBlockQuote = initialText.parent!.parent! as! BlockQuote
-        self.content = Array(newBlockQuote.blockChildren)
+        self.kind = Kind(rawValue: kindTag.tag)!
+        self.content = Array(kindTag.newBlockQuote.blockChildren)
     }
 }
 
 extension BlockQuote {
-    /// Conservatively checks the text of this block quote to see whether it can be parsed as an aside.
-    ///
-    /// Whereas ``Aside/init(_:)`` will use all the text before the first colon in the first line,
-    /// or else return an ``Aside`` with a ``Aside/Kind`` of ``Aside/Kind/note``,
-    /// this function will allow parsers to only parse an aside if there is a single-word aside
-    /// marker in the first line, and otherwise fall back to a plain ``BlockQuote``.
-    func isAside() -> Bool {
+    func parseAsideTag(tagRequirement: Aside.TagRequirement) -> (tag: String, newBlockQuote: BlockQuote)? {
         guard let initialText = self.child(through: [
             (0, Paragraph.self),
             (0, Text.self),
-        ]) as? Text else {
-            return false
+        ]) as? Text, let firstColonIndex = initialText.string.firstIndex(of: ":") else {
+            return nil
         }
 
-        for character in initialText.string {
-            switch character {
-                case ":": return true  // encountered ":" before " "
-                case " ": return false // encountered " " before ":"
-                default:  continue
-            }
+        let kindTag = initialText.string[..<firstColonIndex]
+        let trimmedText = initialText.string[initialText.string.index(after: firstColonIndex)...].drop {
+            $0 == " " || $0 == "\t"
+        }
+
+        guard tagRequirement != .requireSingleWordTag || !kindTag.contains(" ") else {
+            return nil
         }
-        return false // didn't encounter either " " or ":" in the string
+
+        let shiftCount = kindTag.count + 1 + initialText.string[firstColonIndex...].dropFirst().prefix(while: {
+            $0 == " " || $0 == "\t"
+        }).count
+        let textRange: SourceRange? = initialText.range.map({ originalRange in
+            var newStart = originalRange.lowerBound
+            newStart.column += shiftCount
+            return newStart..<originalRange.upperBound
+        })
+
+        guard let newBlockQuote = self._data.substitutingChild(
+            .text(parsedRange: textRange, string: String(trimmedText)),
+            through: [0, 0],
```

Comment:
Question: what does it mean to pass `[0, 0]` here?

Reply:
It's a child path, in this case the first child of the first child. The descendant that is pointed to by this path will be substituted by the given markup data, and then the ancestors of that node will be updated/copied to reflect their new children.